app/graphql/client.tsx

```diff
@@ -69,7 +69,13 @@ const getAuthorizationHeader = (token: string): string => {
 
 const GaloyClient: React.FC<PropsWithChildren> = ({ children }) => {
   const { appConfig, saveToken } = useAppConfig()
-  const [networkError, setNetworkError] = useState<ServerError>()
+
+  const [_, setHasNetworkError] = useState(false)
```

Comment:
I don't understand the purpose of this function

Reply:
Updating a ref variable doesnt trigger a rerender. We need a way to trigger a rerender while still having the ability to set the error only once. If we were to store the networkError in a useState hook, then the client creation uesEffect would have to depend on the networkError which would cause a new client to be recreated whenever the networkError changed. Instead we set the networkErrorRef.current and then trigger a rerender with the setHasNetworkError. I suppose we could invert the two and make `hasNetworkError` be the ref and `setNetworkError` to useState.